From 8608e016d3f3d2d384ccf6e0a7f6ad89dd736462 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Thu, 28 Apr 2022 19:50:01 +0300
Subject: [PATCH] [SW-0] rom: override sku value for onix

Onix FPGA image for gaudi3 supports only single die config. This
patch forces the sku value to single die config. This is required to
prevent fw from executing slave specific logic.

Change-Id: Ic072d7665830b8181c8fdd63f049b242b7540bb0
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index b1bb431..4bb0357 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -385,6 +385,11 @@
 		arcboot_rom_global.config_fuse &= ~EFUSE_SYSCFG_ENABLE_VALIDATION_MASK;
 }
 
+__weak int arcboot_rom_fuse_single_die_config(void)
+{
+	return !!(arcboot_rom_global.config_fuse & EFUSE_SYSCFG_SKU_MASK);
+}
+
 void arcboot_rom_fuse(void)
 {
 	uint32_t val, strap_l;
@@ -398,7 +403,7 @@
 	arcboot_rom_global.config_fuse = val;
 	fuse_read(EFUSE_SYSCFG2_OFFSET, &arcboot_rom_global.config2_fuse);
 
-	if (arcboot_rom_global.config_fuse & EFUSE_SYSCFG_SKU_MASK)
+	if (arcboot_rom_fuse_single_die_config())
 		arcboot_rom_global.sku = 1;
 
 	/* Override the fuse values as per requirement */
diff --git a/arch/gaudi3/arc/onix.c b/arch/gaudi3/arc/onix.c
index 17657e0..24e9515 100644
--- a/arch/gaudi3/arc/onix.c
+++ b/arch/gaudi3/arc/onix.c
@@ -55,3 +55,8 @@
 
 	return 0;
 }
+
+int arcboot_rom_fuse_single_die_config(void)
+{
+	return true;
+}
